maxfw/utils/image_functions.py

```diff
@@ -46,8 +47,12 @@ def to_pil_image(pic, mode=None):
             # if 2D image, add channel dimension (HWC)
             pic = np.expand_dims(pic, 2)
     
-    elif isinstance(pic, io.BytesIO):
-        pic = np.frombuffer(pic)
+    elif isinstance(pic, (bytes, bytearray)):
+        try:
+            # verify that the object can be loaded into memory
+            pic = np.array(Image.open(io.BytesIO(pic)))
```

Comment:
The pic bytes object needs to be converted to `io.BytesIO` first. Then, we'll check if we are able to read it with Pillow, otherwise throw an error.